common/src/main/java/com/wynntils/models/abilities/type/AbilityTreeSkillNode.java

```diff
@@ -0,0 +1,174 @@
+                continue;
+            }
+
+            matcher = text.getMatcher(NODE_BLOCKS_ABILITY_PATTERN);
+            if (matcher.matches()) {
+                blocks.add(matcher.group(1));
+                continue;
+            }
+
+            matcher = text.getMatcher(NODE_REQUIRED_ABILITY_PATTERN);
+            if (matcher.matches()) {
+                requiredAbility = matcher.group(1);
+                continue;
+            }
+
+            matcher = text.getMatcher(NODE_REQUIRED_ARCHETYPE_PATTERN);
+            if (matcher.matches()) {
+                requiredArchetype = new AbilityTreeSkillNode.ArchetypeRequirement(
+                        matcher.group(1), Integer.parseInt(matcher.group(3)));
+                continue;
+            }
+
+            matcher = text.getMatcher(NODE_ARCHETYPE_PATTERN);
+            if (matcher.matches()) {
+                archetype = matcher.group(1);
+                continue;
+            }
+
+            matcher = text.getMatcher(NODE_BLOCKED);
+            if (matcher.matches()) {
+                state = AbilityTreeNodeState.BLOCKED;
+                continue;
+            }
+
+            matcher = text.getMatcher(NODE_UNLOCKED);
+            if (matcher.matches()) {
+                state = AbilityTreeNodeState.UNLOCKED;
+                continue;
+            }
+        }
+
+        AbilityTreeSkillNode.ItemInformation itemInformation = new AbilityTreeSkillNode.ItemInformation(
+                Item.getId(itemStack.getItem()),
+                switch (state) {
+                    case LOCKED -> itemStack.getDamageValue();
+                    case UNLOCKABLE -> itemStack.getDamageValue() - 1;
+                    case UNLOCKED -> itemStack.getDamageValue() - 2;
+                    case BLOCKED -> itemStack.getDamageValue() - 3;
+                });
+
+        AbilityTreeSkillNode node = new AbilityTreeSkillNode(
+                actualName.getString(PartStyle.StyleType.NONE),
+                actualName.getString(PartStyle.StyleType.DEFAULT),
+                loreStyledText.stream()
+                        .map(styledText -> styledText.getString(PartStyle.StyleType.DEFAULT))
+                        .toList(),
+                itemInformation,
+                cost,
+                blocks,
+                requiredAbility,
+                requiredArchetype,
+                archetype,
+                AbilityTreeLocation.fromSlot(slot, page),
+                new ArrayList<>());
+
+        return Pair.of(node, state);
+    }
+
+    public static boolean isNodeItem(ItemStack itemStack, int slot) {
+        StyledText nameStyledText = StyledText.fromComponent(itemStack.getHoverName());
+        return itemStack.getItem() == Items.STONE_AXE
+                && slot < 54
+                && nameStyledText.getMatcher(NODE_NAME_PATTERN).matches();
+    }
+
+    public record ArchetypeRequirement(String name, int required) {}
+
+    public record ItemInformation(int itemId, int damage) {
```

Comment:
It could be worth considering moving these classes to their own files, at least now when the ItemInformation got more complex.